### ISSUE TITLE
Every time the value updates, cursor gets set to the end of the field

### ISSUE THREAD

**Issue by mjp0** [inokawa/rich-textarea#127]
**Describe the bug**
Every time the value updates, cursor gets set to the end of the field.

**To Reproduce**
I'm using valtio as a state manager and `mem_ai` is a proxy store. I tried this with `useState` but didn't change the behavior.

```tsx
<RichTextarea
  id="name"
  rows={1}
  autoHeight
  placeholder="Type something"
  style={{ width: "100%", resize: "none" }}
  value={mem_ai.values.name}
  onChange={(e) => {
    mem_ai.values.name = e.target.value
  }}
/>
```

**Expected behavior**
To be able to write to the field without feeling like I want to break the keyboard ;)

**Platform:**
 - OS: MacOS
 - Browser: Chrome
 - Version of this package: 0.24.4

**Video:**

https://github.com/inokawa/rich-textarea/assets/1594723/119f5fd0-0df2-4e57-ad05-b4b86ca33f0f

**Comment by inokawa** [comment]
Could you make a reproduction?
https://inokawa.github.io/rich-textarea/?path=/story/basics-textarea--one-line-auto-height is very similar situation to you (without `resize:none` style and `autoHeight` prop) but it looks fine.

**Comment by mjp0** [comment]
I just noticed that this is caused by valtio. When I update state in `onChange`, it works as expected. Earlier I first updated valtio which then updated the state through `useEffect`.
```
onChange={(e) => {
  // mem_ai.values.name = e.target.value
  setName(e.target.value)
}}
```
Probably should investigate why valtio is causing the component to behave like this because having to run updates to global state through an extra useState/useEffect step is not very optimal 🤔
